base/math.jl

```diff
@@ -72,6 +77,13 @@ function clamp!(x::AbstractArray, lo, hi)
     x
 end
 
+function clamp!(x::AbstractArray, ::Type{T}) where {T}
```

Comment:
`fill!` is a little different IMO since it's intended to work on the array as a whole, whereas this `clamp!` method is more like implicit vectorization.